test/core/integration/concrete/lockup-dynamic/streamed-amount-of/streamedAmountOf.tree

```diff
@@ -1,12 +1,6 @@
 StreamedAmountOf_Lockup_Dynamic_Integration_Concrete_Test
 └── given STREAMING status
-   ├── given start time in present
```

Comment:
forgot about modifiers

also, since we are declaring the tree branches in terms of the status, wdyt about doing this:

```solidity
modifier runBothCases() {
    vm.warp({ newTimestamp: defaults.START_TIME() });
    _;

    vm.warp({ newTimestamp: defaults.WARP_26_PERCENT() });
    _;
}

function test_GivenSTREAMINGStatus() external givenNotNull givenNotCanceledStream runBothCases {
    // It should return the correct streamed amount.
    uint128 actualStreamedAmount = lockup.streamedAmountOf(defaultStreamId);
    uint128 expectedStreamedAmount = getBlockTimestamp() == defaults.START_TIME() ? 0 : defaults.WITHDRAW_AMOUNT();
    assertEq(actualStreamedAmount, expectedStreamedAmount, "streamedAmount");
}
```

`STREAMING` status means: 
1. `block.timestamp == startTime`
2. `block.timestamp > startTime`

Reply:
Technically works but may I know your concern with the previous approach? Do you want to DRY'ify it using a single modifier? 

I, personally, preferred previous approach where we had clear tree structure and corresponding tests. Our tests are large. To me, it was a no-brainer to review whether the tests are correct or not. Now, it requires some efforts to verify the same since they dont correctly comply with each other. Having said that, I'd personally prefer repeat code in such tests over DRY'ify. And this is why I strongly insist to stick to BTT unless we have find a better framework to follow.

PS: I may be wrong in understanding your reply here. Feel free to correct me if thats the case.

Reply:
> but may I know your concern with the previous approach? 

there were duplicated tests which were testing the same thing, but the test branches were formulated differently

e.g.

in streamed amount test from lockup-base dir:

https://github.com/sablier-labs/v2-core/blob/6a8ec1318f598c351606eb196b9ccebed7dedbab/test/core/integration/concrete/lockup-base/streamed-amount-of/streamedAmountOf.t.sol#L36-L43

in  streamed amount test from lockup-linear dir:

https://github.com/sablier-labs/v2-core/blob/6a8ec1318f598c351606eb196b9ccebed7dedbab/test/core/integration/concrete/lockup-linear/streamed-amount-of/streamedAmountOf.t.sol#L27-L33

---

> it requires some efforts to verify the same since they dont correctly comply with each other

i am sorry, i didn't get this

> 'd personally prefer repeat code in such tests over DRY'ify. And this is why I strongly insist to stick to BTT unless we have find a better framework to follow.

once again, same question from above, why adding other modifiers that have unique logic, non-BTT compatible?